src/zui/ZUICreatePerson/index.tsx

```diff
@@ -0,0 +1,215 @@
+import isEmail from 'validator/lib/isEmail';
+import isURL from 'validator/lib/isURL';
+import { isValidPhoneNumber } from 'libphonenumber-js';
+import {
+  Box,
+  Button,
+  Dialog,
+  Divider,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material';
+import { FC, useEffect, useState } from 'react';
+
+import messageIds from 'zui/l10n/messageIds';
+import { Msg } from 'core/i18n';
+import PersonalInfoForm from './PersonalInfoForm';
+import useCustomFields from 'features/profile/hooks/useCustomFields';
+import useDebounce from 'utils/hooks/useDebounce';
+import { useNumericRouteParams } from 'core/hooks';
+import { ZetkinCreatePerson, ZetkinCustomField } from 'utils/types/zetkin';
+
+interface ZUICreatePersonProps {
+  onClose: () => void;
+  open: boolean;
+}
+export type ShowAllTriggeredType = 'enter' | 'mouse' | 'none';
+
+const ZUICreatePerson: FC<ZUICreatePersonProps> = ({ open, onClose }) => {
+  const theme = useTheme();
+  const { orgId } = useNumericRouteParams();
+  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
+  const customFields = useCustomFields(orgId).data ?? [];
+
+  const customFieldsKeys = customFields.reduce(
+    (acc: { [key: string]: null }, cur: ZetkinCustomField) => {
+      if (cur.type !== 'json') {
+        acc[cur.slug] = null;
+      }
+      return acc;
+    },
+    {}
+  );
+  const initialValue: ZetkinCreatePerson = {
+    ...{
+      alt_phone: null,
+      city: null,
+      co_address: null,
+      country: null,
+      customFields: { ...customFieldsKeys },
+      email: null,
+      ext_id: null,
+      first_name: '',
+      gender: null,
+      last_name: '',
+      phone: null,
+      street_address: null,
+      tags: [],
+      zip_code: null,
+    },
+  };
+  const [personalInfo, setPersonalInfo] =
+    useState<ZetkinCreatePerson>(initialValue);
+
+  useEffect(() => {
+    setPersonalInfo(initialValue);
+  }, [Object.keys(customFieldsKeys).length]);
```

Comment:
When does this happen?

Reply:
After talking about it: It happens when the custom fields load after the UI has rendered, but it would be better to have a loading indicator and not show the UI until everything has loaded (including fields and tags).